src/OctoshiftCLI.Tests/Octoshift/Services/GithubClientTests.cs

```diff
@@ -303,6 +305,76 @@ public async Task PostAsync_Returns_String_Response()
+        var handlerMock = new Mock<HttpMessageHandler>();
+
+        handlerMock
+            .Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),   // Match any HttpRequestMessage
+                ItExpr.IsAny<CancellationToken>())    // Match any CancellationToken
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(expectedResponse, Encoding.UTF8, "application/json")
+            });
+
+        using var httpClient = new HttpClient(handlerMock.Object);
+        var githubClient = new GithubClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, _dateTimeProvider.Object, PERSONAL_ACCESS_TOKEN);
+
+        var stream = new MemoryStream(Encoding.UTF8.GetBytes("Request body content"));
+        var streamContent = new StreamContent(stream);
+        streamContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/json");
+
+        // Act
+        var actualContent = await githubClient.PostAsync("http://example.com", streamContent);
+
+        // Assert
+        actualContent.Should().Be(expectedResponse);
+    }
+
+    [Fact]
+    public async Task PostAsync_With_MultipartFormDataContent_Returns_String_Response()
+    {
+        // Arrange
+        var expectedResponse = "Expected response content";
+        var handlerMock = new Mock<HttpMessageHandler>();
+
+        handlerMock
+            .Protected()
+            .Setup<Task<HttpResponseMessage>>(
+                "SendAsync",
+                ItExpr.IsAny<HttpRequestMessage>(),   // Match any HttpRequestMessage
+                ItExpr.IsAny<CancellationToken>())    // Match any CancellationToken
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent(expectedResponse, Encoding.UTF8, "application/json")
+            });
+
+        using var httpClient = new HttpClient(handlerMock.Object);
+        var githubClient = new GithubClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, _dateTimeProvider.Object, PERSONAL_ACCESS_TOKEN);
+
+        var multipartContent = new MultipartFormDataContent();
+        multipartContent.Add(new StringContent("string content", Encoding.UTF8), "stringPart");
+
+        // Add file content (using a memory stream as a file placeholder)
+        var fileContent = new StreamContent(new MemoryStream(Encoding.UTF8.GetBytes("file content")));
+        fileContent.Headers.ContentType = new System.Net.Http.Headers.MediaTypeHeaderValue("application/octet-stream");
+        multipartContent.Add(fileContent, "filePart", "example.txt");
+
+        // Act
+        var actualContent = await githubClient.PostAsync("http://example.com", multipartContent);
+
+        // Assert
+        actualContent.Should().Be(expectedResponse);
```

Comment:
Same comments apply here too:
```suggestion
        // Arrange
        using var stream = new StreamContent(new MemoryStream(new byte[] { 1, 2, 3 }));
        stream.Headers.ContentType = new("application/octet-stream");
#pragma warning disable IDE0028
        using var expectedMultipartContent = new MultipartFormDataContent();
        expectedMultipartContent.Add(stream, "filePart", "example.txt");
#pragma warning restore

        var handlerMock = MockHttpHandler(req => req.Method == HttpMethod.Post && req.Content == expectedMultipartContent);
        using var httpClient = new HttpClient(handlerMock.Object);
        var githubClient = new GithubClient(_mockOctoLogger.Object, httpClient, null, _retryPolicy, _dateTimeProvider.Object, PERSONAL_ACCESS_TOKEN);

        // Act
        var actualContent = await githubClient.PostAsync("http://example.com", expectedMultipartContent);

        // Assert
        actualContent.Should().Be(EXPECTED_RESPONSE_CONTENT);
```